### PR TITLE
MM-23620: Handle error from GetUser

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

In case of high DB load, the DB will start to throw errors.
Unless we handle the error appropriately, the server will crash.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23620

### CODE REVIEW

File: app/channel_test.go [hunk 1]
```diff
@@ -1648,3 +1651,36 @@ func TestPatchChannelModerationsForChannel(t *testing.T) {
...
+	defer th.TearDown()
+
+	mockStore := th.App.Srv().Store.(*mocks.Store)
+	mockUserStore := mocks.UserStore{}
+	mockUserStore.On("Count", mock.Anything).Return(int64(10), nil)
+	mockUserStore.On("Get", "userID").Return(nil, model.NewAppError("SqlUserStore.Get", "store.sql_user.get.app_error", nil, "user_id=userID", http.StatusInternalServerError))
+	mockPostStore := mocks.PostStore{}
+	mockPostStore.On("GetMaxPostSize").Return(65535, nil)
+	mockSystemStore := mocks.SystemStore{}
+	mockSystemStore.On("GetByName", "InstallationDate").Return(&model.System{Name: "InstallationDate", Value: "10"}, nil)
```
Comment (streamer45) [inline app/channel_test.go:1668]: Do we actually need these for this test?  
I am thinking ChannelStore and UserStore should be sufficient.

Comment (agnivade) [reply app/channel_test.go:1668]: Ah right .. copy pasta.

**Comment by agnivade** [comment]
@ogi-m - This shouldn't need any testing because it's covered by a unit test, and it's pretty hard to trigger it externally, because you need high load with few DB connections available for it to happen. But let me know what you think. Thanks.

**Comment by streamer45** [comment]
> @ogi-m - This shouldn't need any testing because it's covered by a unit test, and it's pretty hard to trigger it externally, because you need high load with few DB connections available for it to happen. But let me know what you think. Thanks.

I guess I could assist with some *peer-testing* :grinning: 
Will run a couple of load-tests on top of this PR and make sure the panic is not happening.

**Comment by ogi-m** [comment]
Thanks Agniva, approved 👍

**Comment by streamer45** [comment]
Ran a couple of one hour long load-tests on top of this PR specifically trying to trigger the panic under high stress conditions. No errors to report. 

Good to merge :+1:

**Comment by agnivade** [comment]
/cherry-pick release-5.23.

Backporting it, as per discussion here: https://community-daily.mattermost.com/core/channels/developers-performance/nhqsnts8k3ge78o3jwu955356w

**Pull request merged**